dql/parser.go

```diff
@@ -1721,8 +1721,14 @@ L:
 							itemInFunc.Errorf("len function only allowed inside inequality" +
 								" function")
 					}
-					function.Attr = nestedFunc.NeedsVar[0].Name
-					function.IsLenVar = true
+					if len(function.Attr) == 0 {
+						// eq(len(a), 1)
+						function.Attr = nestedFunc.NeedsVar[0].Name
+						function.IsLenVar = true
+					} else {
+						// eq(1, len(a))
+						return nil, itemInFunc.Errorf("incorrect order of value and nested function")
```

Comment:
I am still not happy with the error, this is the error that is going to show up in front of the user and he/she may not understand what nested function is.